File: Dante2/Module2.07/Zadanie1.01/main.c
#include <stdio.h>
#include "int_operations.h"
#include "tested_declarations.h"
#include "rdebug.h"

int main()
{
    int error;

    int a, b;
    printf("Enter a und b: ");
    error = scanf("%i %i",&a, &b);
    if(error != 2)
    {
        printf("Incorrect input");
        return 1;
    }

    int type;
    printf("Enter type: ");
    error = scanf("%i",&type);
    if(error != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    switch (type)
    {
        case 0:
            printf("%i", calculate(a, b, add_int));
            break;
        case 1:
            printf("%i", calculate(a, b, sub_int));
            break;
        case 2:
            if(b == 0)
            {
                printf("Incorrect input data");
                return 2;
            }
            printf("%i", calculate(a, b, div_int));
            break;
        case 3:
            printf("%i", calculate(a, b, mul_int));
            break;
        default:
            printf("Incorrect input data");
            return 2;
    }

    return 0;
}

File: Dante2/Module2.07/Zadanie1.01/main.rs
use crate::int_operations::{add_int, calculate, div_int, mul_int, sub_int};

mod int_operations;

fn main() {
    let mut input: String = String::new();

    input.clear();
    println!("Enter a: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let a: i32 = input.trim().parse().expect("Incorrect parse");

    input.clear();
    println!("Enter b: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let b: i32 = input.trim().parse().expect("Incorrect parse");

    input.clear();
    println!("Enter type: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let t: i32 = input.trim().parse().expect("Incorrect parse");

    match t {
        0 => {
            print!("Add: {}", calculate(a, b, add_int));
        },
        1 => {
            print!("Sub: {}", calculate(a, b, sub_int));
        },
        2 => {
            print!("Mul: {}", calculate(a, b, mul_int));
        },
        3 => {
            print!("Div: {}", calculate(a, b, div_int));
        },
        _other => {
            panic!("Incorrect input!");
        }
    }
}
